boa/src/builtins/regexp/mod.rs

```diff
@@ -137,34 +136,26 @@ impl RegExp {
         if regex_flags.contains('i') {
             ignore_case = true;
             sorted_flags.push('i');
-            pattern.push('i');
         }
         if regex_flags.contains('m') {
             multiline = true;
             sorted_flags.push('m');
-            pattern.push('m');
         }
         if regex_flags.contains('s') {
             dot_all = true;
             sorted_flags.push('s');
-            pattern.push('s');
         }
         if regex_flags.contains('u') {
             unicode = true;
             sorted_flags.push('u');
-            //pattern.push('s'); // rust uses utf-8 anyway
         }
         if regex_flags.contains('y') {
             sticky = true;
             sorted_flags.push('y');
         }
-        // the `regex` crate uses '(?{flags})` inside the pattern to enable flags
-        if !pattern.is_empty() {
-            pattern = format!("(?{})", pattern);
-        }
-        pattern.push_str(regex_body.as_str());
 
-        let matcher = Regex::new(pattern.as_str()).expect("failed to create matcher");
+        let matcher = Regex::newf(regex_body.as_str(), Flags::from(sorted_flags.as_str()))
+            .expect("failed to create matcher");
```

Comment:
I don't think a panic here is spec-compliant. We should never panic if we get an invalid regular expression. I know this was happening before, but It's best if we can avoid it, now that we are working on this.

Could we implement the constructor in compliance with the spec?
https://tc39.es/ecma262/#sec-regexp-constructor

I don't think we should block the PR on this, but if it's merged as-is, we should create an issue about this.

Reply:
I'd say to open an issue mentioning lack of spec compliability for `RegExp`s, there's a bit of work to be done in both `boa`'s side and `regress`'s.

Reply:
Let's do that :)